File: android/app/src/main/java/tech/loveace/appv3/data/model/LoginStatus.kt
package tech.loveace.appv3.data.model

data class ECLoginStatus(
    val success: Boolean = false,
    val failNotFoundTwfid: Boolean = false,
    val failNotFoundRsaKey: Boolean = false,
    val failNotFoundRsaExp: Boolean = false,
    val failNotFoundCsrfCode: Boolean = false,
    val failInvalidCredentials: Boolean = false,
    val failMaybeAttacked: Boolean = false,
    val failNetworkError: Boolean = false,
    val failUnknownError: Boolean = false,
) {
    val errorMessage: String
        get() = when {
            failNotFoundTwfid -> "未找到TwfID"
            failNotFoundRsaKey -> "未找到RSA密钥"
            failNotFoundRsaExp -> "未找到RSA指数"
            failNotFoundCsrfCode -> "未找到CSRF代码"
            failInvalidCredentials -> "用户名或密码错误"
            failMaybeAttacked -> "登录频繁，请稍后重试"
            failNetworkError -> "网络连接错误"
            failUnknownError -> "未知错误"
            else -> ""
        }
}

data class UAAPLoginStatus(
    val success: Boolean = false,
    val failNotFoundLt: Boolean = false,
    val failNotFoundExecution: Boolean = false,
    val failInvalidCredentials: Boolean = false,
    val failNetworkError: Boolean = false,
    val failUnknownError: Boolean = false,
) {
    val errorMessage: String
        get() = when {
            failNotFoundLt -> "未找到lt参数"
            failNotFoundExecution -> "未找到execution参数"
            failInvalidCredentials -> "用户名或密码错误"
            failNetworkError -> "网络连接错误"
            failUnknownError -> "未知错误"
            else -> ""
        }
}

File: ios/loveaceios/Models/LoginStatus.swift
import Foundation

struct ECLoginStatus {
    let success: Bool
    let failNotFoundTwfid: Bool
    let failNotFoundRsaKey: Bool
    let failNotFoundRsaExp: Bool
    let failNotFoundCsrfCode: Bool
    let failInvalidCredentials: Bool
    let failMaybeAttacked: Bool
    let failNetworkError: Bool
    let failUnknownError: Bool

    var errorMessage: String {
        if failNotFoundTwfid { return "未找到TwfID" }
        if failNotFoundRsaKey { return "未找到RSA密钥" }
        if failNotFoundRsaExp { return "未找到RSA指数" }
        if failNotFoundCsrfCode { return "未找到CSRF代码" }
        if failInvalidCredentials { return "用户名或密码错误" }
        if failMaybeAttacked { return "登录频繁，请稍后重试" }
        if failNetworkError { return "网络连接错误" }
        if failUnknownError { return "未知错误" }
        return ""
    }

    init(success: Bool = false, failNotFoundTwfid: Bool = false,
         failNotFoundRsaKey: Bool = false, failNotFoundRsaExp: Bool = false,
         failNotFoundCsrfCode: Bool = false, failInvalidCredentials: Bool = false,
         failMaybeAttacked: Bool = false, failNetworkError: Bool = false,
         failUnknownError: Bool = false) {
        self.success = success; self.failNotFoundTwfid = failNotFoundTwfid
        self.failNotFoundRsaKey = failNotFoundRsaKey; self.failNotFoundRsaExp = failNotFoundRsaExp
        self.failNotFoundCsrfCode = failNotFoundCsrfCode
        self.failInvalidCredentials = failInvalidCredentials
        self.failMaybeAttacked = failMaybeAttacked; self.failNetworkError = failNetworkError
        self.failUnknownError = failUnknownError
    }
}

struct UAAPLoginStatus {
    let success: Bool
    let failNotFoundLt: Bool
    let failNotFoundExecution: Bool
    let failInvalidCredentials: Bool
    let failNetworkError: Bool
    let failUnknownError: Bool

    var errorMessage: String {
        if failNotFoundLt { return "未找到lt参数" }
        if failNotFoundExecution { return "未找到execution参数" }
        if failInvalidCredentials { return "用户名或密码错误" }
        if failNetworkError { return "网络连接错误" }
        if failUnknownError { return "未知错误" }
        return ""
    }

    init(success: Bool = false, failNotFoundLt: Bool = false,
         failNotFoundExecution: Bool = false, failInvalidCredentials: Bool = false,
         failNetworkError: Bool = false, failUnknownError: Bool = false) {
        self.success = success; self.failNotFoundLt = failNotFoundLt
        self.failNotFoundExecution = failNotFoundExecution
        self.failInvalidCredentials = failInvalidCredentials
        self.failNetworkError = failNetworkError; self.failUnknownError = failUnknownError
    }
}
